docs/docker-stack/docker-images-recipes/go-beam.Dockerfile

```diff
@@ -0,0 +1,35 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+ARG BASE_AIRFLOW_IMAGE
+FROM ${BASE_AIRFLOW_IMAGE}
+
+SHELL ["/bin/bash", "-o", "pipefail", "-e", "-u", "-x", "-c"]
+
+USER 0
+
+ENV GO_INSTALL_DIR=/usr/local/go
+
+# Install Go
+RUN DOWNLOAD_URL="https://dl.google.com/go/go1.16.4.linux-amd64.tar.gz" \
+    && TMP_DIR="$(mktemp -d)" \
+    && curl -fL "${DOWNLOAD_URL}" --output "${TMP_DIR}/go.linux-amd64.tar.gz" \
+    && mkdir -p "${GO_INSTALL_DIR}" \
+    && tar xzf "${TMP_DIR}/go.linux-amd64.tar.gz" -C "${GO_INSTALL_DIR}" --strip-components=1 \
+    && rm -rf "${TMP_DIR}"
+
+ENV GOROOT=/usr/local/go
+ENV PATH="$GOROOT/bin:$PATH"
```

Comment:
Can you add a check if the application is available and if not, raise an exception with a link to this documentation and golang website?
We should add check here:
https://github.com/apache/airflow/blob/774ca085d2d2d58d4292b43c3511a145cc07154b/airflow/providers/apache/beam/hooks/beam.py#L310
To check if program exists, we should use [`shutil.which`](https://docs.python.org/3/library/shutil.html#shutil.which)
To generate link, we shouldn't use `airflow.airflow.utils.docs`, because this doc is not versioned. We can use the static link.

Reply:
Good idea. I just added that check and a test as well to validate this behavior.